doc/api/stream.md

```diff
@@ -540,16 +540,18 @@ possible to respect backpressure and avoid memory issues using the
 [`'drain'`][] event:
 
 ```js
-function write(data, cb) {
-  if (!stream.write(data)) {
-    stream.once('drain', cb);
-  } else {
-    process.nextTick(cb);
+async function waitForDrain(stream) {
+  return await new Promise(resolve => stream.once('drain', resolve));
```

Comment:
If I understand correctly, we do not need to `await` here, otherwise, this will be [extra promise rewrapping](http://exploringjs.com/es2016-es2017/ch_async-functions.html#_returned-promises-are-not-wrapped).

Reply:
@vsemozhetbyt well, agruably it produces a nicer stack-trace.

Reply:
That, basically. I could add comments, but I feel it would distract from the real purpose of the example, which is to show how to wait for drain to complete.